src/main/java/dev/sigstore/rekor/client/RekorClient.java

```diff
@@ -88,16 +91,69 @@ public RekorResponse putEntry(HashedRekordRequest hashedRekordRequest) throws IO
     if (resp.getStatusCode() != 201) {
       throw new IOException(
           String.format(
-              "bad response from rekor @ '%s' : %s", rekorEndpoint, resp.parseAsString()));
+              "bad response from rekor @ '%s' : %s", rekorPutEndpoint, resp.parseAsString()));
     }
 
     URI rekorEntryUri = serverUrl.resolve(resp.getHeaders().getLocation());
     String entry = resp.parseAsString();
     return RekorResponse.newRekorResponse(rekorEntryUri, entry);
   }
 
-  /** Obtain an entry for an artifact from rekor. */
-  public void getEntry() {
-    throw new UnsupportedOperationException("I'm a worthless download function");
+  public RekorEntry getEntry(String UUID) throws IOException {
+    URI getEntryURI = serverUrl.resolve(REKOR_ENTRIES_GET_PATH + UUID);
+    HttpRequest req =
+        httpProvider
+            .getHttpTransport()
+            .createRequestFactory()
+            .buildGetRequest(new GenericUrl(getEntryURI));
+    req.getHeaders().set("Accept", "application/json");
+    HttpResponse response;
+    try {
+      response = req.execute();
+    } catch (HttpResponseException e) {
+      if (e.getStatusCode() == 404) return null;
+      throw e;
+    }
+    return RekorResponse.newRekorResponse(getEntryURI, response.parseAsString()).getEntry();
+  }
+
+  /**
+   * Returns a list of UUIDs for matching entries for the given search parameters.
+   *
+   * @param email the OIDC email subject
+   * @param hash sha256 hash of the artifact
+   * @param publicKeyFormat format of public key (one of 'pgp','x509','minisign', 'ssh', 'tuf')
+   * @param publicKeyContent public key base64 encoded content
+   */
+  public List<String> searchEntry(
```

Comment:
Is the workflow supposed to be?
1. search of UUID
2. get entry by UUID

Reply:
That's what the web UI does, it seems like the only way to retrieve a full rekord without a uuid. That doesn't mean we can't present a nicer API to clients.

Reply:
Word. I didn't realize that's how it worked